src/wallet/wallet2.cpp

```diff
@@ -480,7 +480,10 @@ size_t estimate_rct_tx_size(int n_inputs, int mixin, int n_outputs, size_t extra
   size += 1;
 
   // rangeSigs
-  size += (2*64*32+32+64*32) * n_outputs;
+  if (bulletproof)
+    size += (704 + 3) * n_outputs;
+  else
+    size += (2*64*32+32+64*32) * n_outputs;
```

Comment:
AIUI, for the Borromean case, there are 2*64 +1 scalars (`s0, s1, ee` in `boroSig`) and 64 points (`Ci` in `rangeSig`) that make up this number. What would be the breakdown for BPs?

The paper states at the end of Section 4.2 that 

> In total the prover sends `2*log_2(n)+4` group elements and 5 elements in Zp.

So these 4 group elements and 5 scalar elements must be `A, S, T1, T2` and `taux, mu, a, b, t` in `Bulletproof`, respectively. And we have `n=64` so `2*log_2(n) = 12`. So my calculation would be `32*(12 + 9) = 672`. Where am I wrong?

Reply:
Maybe it's more helpful to just keep the breakdown as in the paper: `(2*64 + 4 +5)*32`

Reply:
I had counted V too, but I don't save it anymore, so this needs updating, yes.